packages/tooling/fast-tooling/src/data-utilities/shortcut-action.ts

```diff
@@ -0,0 +1,107 @@
+export interface MetaKey {
+    metaKey: true;
+}
+
+export interface ShiftKey {
+    shiftKey: true;
+}
+
+export interface CtrlKey {
+    ctrlKey: true;
+}
+
+export interface AltKey {
+    altKey: true;
+}
+
+export type ModifierKey = AltKey | CtrlKey | ShiftKey | MetaKey;
+
+export interface SpecificKey {
+    /**
+     * The key string value
+     */
+    value: string;
+}
+
+export type KeyConfig = SpecificKey | ModifierKey;
+
+export interface ShortcutActionConfig {
+    /**
+     * The display name of the shortcut action
+     */
+    name: string;
+
+    /**
+     * The keys needed to execute the action
+     */
+    keys: KeyConfig[];
+
+    /**
+     * The action to take when the keycodes have been pressed
+     */
+    action: () => void;
```

Comment:
You might consider supplying the action with the config. Doing so would allow authors more flexibility in how they provide callbacks to the system because they can perform delegation in their own code if they prefer to not supply unique functions to each action config.